cowprotocol/cowswap
# [4] feat(balances): update balances for priority tokens

# Summary

Fixes: #3430, #3429, #3426, #3423, #3428

Since the regular balances+allowances updating polling happens only every ~30 seconds, it might be annoying if you order was just filled and you don't see balances updater. Or you just approved your token spending and still see "Approve token" state.

To fix that, I've created `PriorityTokensUpdater` that is almost the same as `BalancesAndAllowancesUpdater`.
The only difference is that the priority updater updates only assets from the current trade state + from pending orders. And it does it quite often - every ~10 seconds.

  # To Test

1. Create an order
2. Wait until it is filled
- [ ] traded assets balances should be updated within ~10 seconds
3. Approve some token
- [ ] the "Approve token" state should disappear within ~10 seconds
